server/postgresDB/insert.go

```diff
@@ -1,119 +1,30 @@
 package postgresDB
 
 import (
-	"database/sql"
+	"context"
+	dataStruct "devashishRaj/rpi_telemetry/server/dataStruct"
 	"fmt"
 	"log"
-	dataStruct "server/dataStruct"
 
-	_ "github.com/lib/pq"
+	_ "github.com/jackc/pgx/v5"
 )
 
-func CheckDevicesDB(jsonData dataStruct.SystemInfo) {
-	db = ConnectDB()
-	query := (`
-		SELECT EXISTS (SELECT macaddress 
-			  FROM telemetry.devices 
-			WHERE  macaddress = $1 );
-		`)
-	var isPresent bool
-	err := db.QueryRow(query, jsonData.MacAddress).Scan(&isPresent)
-
-	if err != nil {
-		fmt.Println("Query error in CheckDevicesDB")
-		log.Fatalln(err)
-
+func InsertInDB(jsonData dataStruct.MetricsBatch) {
+	deviceMetrics := jsonData.Metrics
+	if len(deviceMetrics) == 0 {
+		log.Fatal("jsondata is emply")
 	}
-	fmt.Println("isPresent: ", isPresent)
-	if isPresent == false {
-		_, err := db.Exec(`
-		INSERT INTO telemetry.devices (MacAddress , privateIP ,  publicIP , hostname , 
-		ostype , totalmemory)
-		VALUES ($1, $2, $3, $4, $5, $6)`,
-			jsonData.MacAddress, jsonData.PrivateIP, jsonData.PublicIP,
-			jsonData.Hostname, jsonData.OsType, jsonData.Metrics.TotalMemory)
-		if err != nil {
-			fmt.Println("Query error in CheckDevicesDB when  ")
-			log.Fatalln(err)
 
+	for _, metric := range deviceMetrics {
+		_, err := G_dbpool.Exec(context.Background(), `
+            INSERT INTO telemetry.metrics_new (macaddress, name, value, timestamp)
```

Comment:
future optimisation : instead of doing multiple inserts do a bulk insert using `Copy` protocol. https://pkg.go.dev/github.com/jackc/pgx#hdr-Copy_Protocol

Add a todo for later